src/software/networking/udp/network_utils.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <ifaddrs.h>
+
+#include <string>
+
+/**
+ * Given an interface, get the IP address associated with that interface
+ *
+ * The modified ip_address is valid only if the function returns true
+ *
+ * @param interface The interface to get the IP address from
+ * @param ip_address A reference to the std::string that will store the IP address if
+ * found
+ * @param ipv4 If true, get the IPv4 address, otherwise get the IPv6 address
+ *
+ * @return true if the IP address was found, false otherwise
+ */
+bool getLocalIp(const std::string& interface, std::string& ip_address, bool ipv4 = true);
```

Comment:
Could return `std::optional<std::string>` to avoid the out parameter

Reply:
Good call, updated